onnxruntime/core/profile/context.h

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <thread>
+#include <mutex>
+#include <string>
+#include <unordered_map>
+
+namespace onnxruntime {
+namespace profile {
+
+class Context {
+ public:
+  static Context& GetInstance() {
+    static Context instance_;
+    return instance_;
+  }
+
+  std::string GetThreadTag(const std::thread::id thread_id) {
+    const std::lock_guard<std::mutex> lock(mtx_);
+    return thread_tag_.at(thread_id);
+  }
+
+  void SetThreadTag(
+      const std::thread::id thread_id, const std::string tag) {
```

Comment:
>const std::string [](start = 39, length = 17)

const std::string&

Reply:
Ok.

---
In reply to: [428995106](https://github.com/microsoft/onnxruntime/pull/4013#discussion_r428995106) [](ancestors = 428995106)